### PR TITLE
BUG: Scale PDF annotations

### LINKED ISSUE #1386: Transformation for comment does not work
Replace this: What happened? What were you trying to achieve?

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
# Windows-10-10.0.19044-SP0

$ python -c "import PyPDF2;print(PyPDF2.__version__)"
# 2.2.0
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
from PyPDF2 import PdfReader, PdfWriter, Transformation

reader = PdfReader('check.pdf')
writer = PdfWriter()
for page in reader.pages:
    page.add_transformation(Transformation().scale(0.5))
    writer.add_page(page)
writer.write('output.pdf')
```

Share here the PDF file(s) that cause the issue. The smaller they are, the
better. Let us know if we may add them to our tests!

[check.pdf](https://github.com/py-pdf/PyPDF2/files/9740564/check.pdf)

## Traceback

This is the complete Traceback I see:

[output.pdf](https://github.com/py-pdf/PyPDF2/files/9740565/output.pdf)

By doing a scale with a factor of 0.5, I hope to scale the whole page, but the comments do not scale.

Source:
image

Result:
image

Hope:
image

### PR DESCRIPTION
**Pull request by joshhendo**
Fixes #1386 

PDF annotations - for example hyperlinks and comments - aren't properly scaled when using the scale function.

I tried to follow a similar pattern to the code just below my change which scales the Viewport.

As a test, the input PDF is as follows:
image

In this screenshot, I am clicking on the link and Adobe Acrobat shows it as a black box. This is in the position I expect.

When scaled to 0.9 without this fix, the following occurs:
image

The clickable area for the link has changed, as well as the comment.

The PDF used to test was:
[test_input.pdf](https://github.com/py-pdf/PyPDF2/files/10194119/test_input.pdf)

The PDF scaled to 0.9 before the fix:
[test_output_before.pdf](https://github.com/py-pdf/PyPDF2/files/10194157/test_output_before.pdf)

The PDF scaled to 0.9 after the fix:
[test_output_after.pdf](https://github.com/py-pdf/PyPDF2/files/10194167/test_output_after.pdf)

### CODE REVIEW

**Comment by joshhendo** [comment]
I'm not sure why the CI is failing indicating that the annotations aren't iterable - it seems to be when I run it and I based the code on the docs here: https://pypdf2.readthedocs.io/en/latest/user/reading-pdf-annotations.html?highlight=annotations 

Perhaps it's not guaranteed to be iterable and there needs to be some form of type guard before attempting to iterate it.

File: PyPDF2/_page.py [hunk 1]
```diff
@@ -1057,6 +1058,19 @@ def scale(self, sx: float, sy: float) -> None:
...
+            annotations = self[PG.ANNOTS]
+            for annotation in annotations:
+                annotation_obj = annotation.get_object()
+                if ADA.Rect in annotation_obj:
+                    rectangle = annotation_obj[ADA.Rect]
+                    if isinstance(rectangle, ArrayObject):
+                        rectangle[0] = FloatObject(float(rectangle[0]) * sx)
+                        rectangle[1] = FloatObject(float(rectangle[1]) * sy)
+                        rectangle[2] = FloatObject(float(rectangle[2]) * sx)
+                        rectangle[3] = FloatObject(float(rectangle[3]) * sy)
```
Comment (MartinThoma) [inline PyPDF2/_page.py:1072]:
```suggestion
            if isinstance(annotations, ArrayObject):
                for annotation in annotations:
                    annotation_obj = annotation.get_object()
                    if ADA.Rect in annotation_obj:
                        rectangle = annotation_obj[ADA.Rect]
                        if isinstance(rectangle, ArrayObject):
                            rectangle[0] = FloatObject(float(rectangle[0]) * sx)
                            rectangle[1] = FloatObject(float(rectangle[1]) * sy)
                            rectangle[2] = FloatObject(float(rectangle[2]) * sx)
                            rectangle[3] = FloatObject(float(rectangle[3]) * sy)
```

**Comment by MartinThoma** [comment]
@joshhendo Thank you for taking care of it :pray: I'll release it to PyPI today :-) 

If you want, I can add you to the list of contributors: https://pypdf2.readthedocs.io/en/latest/meta/CONTRIBUTORS.html :-)

**Comment by joshhendo** [comment]
Thanks @MartinThoma ! And for fixing the mypy issue - I was on the right path thinking that narrowing/a type guard was needed. I have never used mypy before so I had been trying to get it working locally for me to work on.

**Comment by joshhendo** [comment]
> If you want, I can add you to the list of contributors: https://pypdf2.readthedocs.io/en/latest/meta/CONTRIBUTORS.html :-)

That would be great!

**Comment by pubpub-zz** [comment]
@joshhendo  / @MartinThoma 
This PR only fixes scaling but not the other transformations. Actually, the annotations reposition should be applied within the `add_transformation` function. 
example with a rotation of -10deg
image

@joshhendo, are you OK to upgrade your code change ?

**Pull request merged**